lib/OpenQA/Schema/Result/JobModules.pm

```diff
@@ -83,14 +83,27 @@ __PACKAGE__->belongs_to(
         on_update     => "CASCADE",
     },
 );
-__PACKAGE__->has_many(needle_hits => 'OpenQA::Schema::Result::JobModuleNeedles', 'job_module_id');
 
 sub sqlt_deploy_hook {
     my ($self, $sqlt_table) = @_;
 
     $sqlt_table->add_index(name => 'idx_job_modules_result', fields => ['result']);
 }
 
+# override to straighten out needle references
+sub delete {
+    my ($self) = @_;
+
+    my $schema = $self->result_source->schema;
+
+    my $needles = $schema->resultset('Needles');
+
+    $needles->search({last_seen_module_id    => $self->id})->update({last_seen_module_id    => undef});
```

Comment:
So you need this despite the `on_delete => 'SET NULL'` above?  
(same for the next line)

Reply:
Yes. I think the on_delete is toothless

Reply:
I tried without the function now - possibly my schema wasn't properly setup when I tried